rust/connlib/clients/shared/src/control.rs

```diff
@@ -46,13 +51,42 @@ pub struct ControlPlane<CB: Callbacks> {
     pub tunnel: Arc<Tunnel<ControlSignaler, CB, ClientState>>,
     pub control_signaler: ControlSignaler,
     pub tunnel_init: Mutex<bool>,
+    // It's a Mutex<Option<_>> because we need the init message to initialize the resolver
+    // also, in platforms with split DNS and no configured upstream dns this will be None.
+    //
+    // We could still initialize the resolver with no nameservers in those platforms...
+    pub fallback_resolver: parking_lot::Mutex<Option<TokioAsyncResolver>>,
```

Comment:
I thought about doing it that way but given the current refactor you are doing I thought this would look better, since:
* In the future holding it in the control plane would allow us to get rid of the `Mutex<Option<_>>` since we can wait for the `init` message to create it and send it as part of `EventLoop::new`
* I think it's better if we control how async tasks are handled from upper layers and leave the tunnel crate as simple as possible